tests/components/modbus/test_modbus_bit_sensor.py

```diff
@@ -0,0 +1,127 @@
+"""The tests for the Modbus sensor component."""
+import pytest
+
+from homeassistant.components.modbus.const import (
+    CALL_TYPE_REGISTER_HOLDING,
+    CALL_TYPE_REGISTER_INPUT,
+    CONF_BIT_NUMBER,
+    CONF_BIT_SENSORS,
+    CONF_COUNT,
+    CONF_INPUT_TYPE,
+    CONF_REGISTERS,
+)
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICE_CLASS,
+    CONF_NAME,
+    CONF_SLAVE,
+    STATE_UNAVAILABLE,
+)
+
+from .conftest import base_config_test, base_test
+
+
+@pytest.mark.parametrize("do_options", [False, True])
+@pytest.mark.parametrize(
+    "do_type", [CALL_TYPE_REGISTER_HOLDING, CALL_TYPE_REGISTER_INPUT]
+)
+async def test_config_sensor(hass, do_options, do_type):
+    """Run test for sensor."""
+    sensor_name = "test_sensor"
+    config_sensor = {CONF_NAME: sensor_name, CONF_ADDRESS: 51, CONF_BIT_NUMBER: 5}
+    if do_options:
+        config_sensor.update(
```

Comment:
Please see in the other test files. if/else in the test cases are to be avoided

Reply:
looks like it's common practice
https://github.com/home-assistant/core/blob/dev/tests/components/modbus/test_modbus_sensor.py#L46
https://github.com/home-assistant/core/blob/dev/tests/components/modbus/test_modbus_switch.py#L62

Reply:
That is old standard, as I noted somewhere else...when you present new code, maintainers look with the present standards. I was asked to remove specifically the “if do_options” in my last PR (and it do make sense). This is a good example why not to move old code.

When I get time, I will change all tests to follow this new pattern.

Reply:
Oh, cool. Sorry, that's my first PR to the HA, I might not know all the rules